swfiles/sw_instrument.m

```diff
@@ -274,27 +274,33 @@
 
 if FX > 0
     k0 = param.k;
-    cosT = cosd(param.thetaMin);
-    sinT = sind(param.thetaMin);
-    
+    cosTMin = cosd(param.thetaMin);
+    sinTMin = sind(param.thetaMin);
+    cosTMax = cosd(param.thetaMax);
+    sinTMax = sind(param.thetaMax);
     
     for jj = 1:nPlot
         switch FX
             case 1
                 % fix ki
-                Emax = (k0^2-(k0*cosT-sqrt(Q.^2-k0^2*sinT^2)).^2) * sw_converter(1,'k','meV');
-                Emin = (k0^2-(k0*cosT+sqrt(Q.^2-k0^2*sinT^2)).^2) * sw_converter(1,'k','meV');
+                % Ross Stewart, 9/2/23 - get proper trajectories based on thetamin *and* thetamax
+                Emax = Q*0;
+                Emin = Q*0;
+                ii = find(Q < k0);
+                Emax(ii) = (k0^2-(k0*cosTMin-sqrt(Q(ii).^2-k0^2*sinTMin^2)).^2) * sw_converter(1,'k','meV');
+                Emin(ii) = (k0^2-(k0*cosTMin+sqrt(Q(ii).^2-k0^2*sinTMin^2)).^2) * sw_converter(1,'k','meV');
+                ii = find(Q > k0);
+                Emax(ii) = (k0^2-(k0*cosTMax+sqrt(Q(ii).^2-k0^2*sinTMax^2)).^2) * sw_converter(1,'k','meV');
+                Emin(ii) = (k0^2-(k0*cosTMax-sqrt(Q(ii).^2-k0^2*sinTMax^2)).^2) * sw_converter(1,'k','meV');
             case 2
                 % fix kf
                 Emax = -(k0^2-(k0*cosT+sqrt(Q.^2-k0^2*sinT^2)).^2) * sw_converter(1,'k','meV');
```

Comment:
This is nit-picking but can we not use `k0` - I think people will assume `k0` is $k_i$ (though I know it's obvious from the comment and the code) ! Can we just call in `kfix` or similar?